wallet/src/de/schildbach/wallet/transactions/WalletObserver.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.transactions
+
+import android.os.Looper
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import org.bitcoinj.core.Context
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.core.listeners.TransactionConfidenceEventListener
+import org.bitcoinj.utils.Threading
+import org.bitcoinj.wallet.Wallet
+import org.bitcoinj.wallet.listeners.WalletChangeEventListener
+import org.bitcoinj.wallet.listeners.WalletCoinsReceivedEventListener
+import org.bitcoinj.wallet.listeners.WalletCoinsSentEventListener
+import org.dash.wallet.common.transactions.filters.TransactionFilter
+
+class WalletObserver(private val wallet: Wallet) {
+    fun observeWalletChanged(): Flow<Unit> = callbackFlow {
+        val walletChangeListener = WalletChangeEventListener {
+            trySend(Unit)
+        }
+
+        wallet.addChangeEventListener(Threading.USER_THREAD, walletChangeListener)
```

Comment:
`USER_THREAD` doesn't make much of a difference in the case of a simple `trySend` since it's quite fast. However, it does make a difference below in the `observeTransactions` method since the transaction filter is running on that thread.

Additionally, I figure if there are any errors caused by `trySend` or the filters, we don't want to crash the thread where the sync is running.